pinot-common/src/main/java/org/apache/pinot/common/function/scalar/DataTypeConversionFunctions.java

```diff
@@ -41,6 +41,9 @@ private DataTypeConversionFunctions() {
 
   @ScalarFunction
   public static Object cast(Object value, String targetTypeLiteral) {
+    if (value == null) {
```

Comment:
We shouldn't need to handle null here if the scalar function doesn't have `nullableParameters`